src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+    const getNonceByHash = async () => {
+      const blockHash = await targetApi.rpc.chain.getBlockHash(bestBlockFinalizedOnTarget);
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(
+        latestReceivedNonceMethodName,
+        laneId,
+        blockHash.toJSON()
+      );
+
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      const latestReceivedNonce = latestReceivedNonceCallType.toString();
+      setNonceOfTargetFinalizedBlock(parseInt(latestReceivedNonce));
+    };
+
+    const getLatestReceivedNonce = async () => {
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(latestReceivedNonceMethodName, laneId);
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      setLatestReceivedNonceRuntimeApi(parseInt(latestReceivedNonceCallType.toString()));
+    };
+
+    getNonceByHash();
+    getLatestReceivedNonce();
+  }, [
+    areApiLoading,
+    bestBlockFinalizedOnTarget,
+    currentTransaction,
+    laneId,
+    latestReceivedNonceMethodName,
+    targetApi.registry,
+    targetApi.rpc.chain,
+    targetApi.rpc.state
+  ]);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: number | null) => {
+      if (!transactionValue || !chainValue) {
+        return false;
+      }
+      return chainValue > transactionValue;
+    };
+
+    const isDone = (status: boolean | boolean) => (currentTransaction.block && status ? 'DONE' : 'RUNNING');
+    const step1 = currentTransaction.block ? `included in block ${currentTransaction.block}` : isDone(false);
+    const step2 = stepEvaluator(currentTransaction.block, parseInt(bestBlockFinalized));
+    const step3 = stepEvaluator(currentTransaction.block, parseInt(importedHeaders));
+    const step4 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceRuntimeApi);
+    const step5 = stepEvaluator(currentTransaction.messageNonce, nonceOfTargetFinalizedBlock);
+    const step6 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceOnSource);
+
+    const steps = [
+      { chainType: sourceChain, label: 'Including message in block', status: step1 },
+      { chainType: sourceChain, label: 'Waiting for block finality on source', status: isDone(step2) },
+      { chainType: targetChain, label: 'Source Chain block finality received on target', status: isDone(step3) },
+      { chainType: targetChain, label: 'Message delivered', status: isDone(step4) },
+      { chainType: targetChain, label: 'Message finality', status: isDone(step5) },
+      { chainType: sourceChain, label: 'Delivery confirmation on source', status: isDone(step6) }
+    ];
```

Comment:
If this was made a component this would simply be a list of components as well, I don't feel like abstracting this is required.